diff --git a/src/Http/Controllers/AuthorizationController.php b/src/Http/Controllers/AuthorizationController.php
index 2123cbe..31866db 100644
--- a/src/Http/Controllers/AuthorizationController.php
+++ b/src/Http/Controllers/AuthorizationController.php
@@ -79,6 +79,11 @@ class AuthorizationController
         $authRequest->setUser(new User($user->getAuthIdentifier()));
 
         $scopes = $this->parseScopes($authRequest);
+
+        if (! $user->hasScopes($scopes)) {
+            return $this->promptForScopes($request);
+        }
+
         $client = $this->clients->find($authRequest->getClient()->getIdentifier());
 
         if ($prompt->doesntContain('consent') &&
@@ -167,4 +172,14 @@ class AuthorizationController
 
         throw new AuthenticationException(guards: isset($this->guard->name) ? [$this->guard->name] : []);
     }
+
+    protected function promptForScopes($request) {
+        $intended = $request->isMethod('GET') && $request->route() && ! $request->expectsJson()
+            ? url()->full()
+            : url()->previous();
+
+        return redirect()
+            ->setIntendedUrl($intended)
+            ->route('cloud.create');
+    }
 }
